packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -1,19 +1,87 @@
 import React, { useState, useEffect, useCallback, useRef } from "react"
-
 import getBuildInfo from "../utils/getBuildInfo"
+import IndicatorButton from "./IndicatorButton"
+import getGatsbyIndicatorButtonProps from "./getGatsbyIndicatorButtonProps"
+import getLinkIndicatorButtonProps from "./getLinkIndicatorButtonProps"
+import getInfoIndicatorButtonProps from "./getInfoIndicatorButtonProps"
+import { gatsbyIcon, infoIcon, linkIcon, successIcon } from "./icons"
 import Style from "./Style"
 
-import GatsbyIndicatorButton from "./GatsbyIndicatorButton"
-import LinkIndicatorButton from "./LinkIndicatorButton"
-import InfoIndicatorButton from "./InfoIndicatorButton"
-
 const POLLING_INTERVAL = process.env.GATSBY_PREVIEW_POLL_INTERVAL || 3000
 
+export function PreviewIndicator({
+  children,
+  gatsbyIndicatorButtonProps,
+  linkIndicatorButtonProps,
+  infoIndicatorButtonProps,
```

Comment:
I'm unsure what these props would be? Wouldn't we better move to a more composable API? In cloud loader you do the following:

```js
<PreviewIndicatorSideBar>
  <GatsbyIndicaor {...gatsbyIndicatorButtonProps} />
  <LinkIndicaor {...linkIndicatorButtonProps} />
  <InfoIndicaor {...infoIndicatorButtonProps} />
</PreviewIndicatorSideBar>
```

It adds more flexibility to the component and follows React's composition philosophy.